test/testautomation_events.c

```diff
@@ -62,10 +69,25 @@ static int events_pushPumpAndPollUserevent(void *arg)
     SDL_PumpEvents();
     SDLTest_AssertPass("Call to SDL_PumpEvents()");
 
-    /* Poll for user event */
-    result = SDL_PollEvent(&event2);
-    SDLTest_AssertPass("Call to SDL_PollEvent()");
-    SDLTest_AssertCheck(result == 1, "Check result from SDL_PollEvent, expected: 1, got: %d", result);
+    /* Poll until we get a user event. */
+    for (;;) {
+        SDL_zero(event2);
```

Comment:
The zero isn't necessary. The event is fully overwritten on successful return.

Reply:
I did it to have the event in a known (zeroed) state in the case where `SDL_PollEvent` is broken and to avoid the event struct containing random data from the previous call to `SDL_PollEvent` (assuming another thread pushed an event in the small time window between `SDL_FlushEvents` and `SDL_PollEvents`).

Reply:
Removed it. The event type will error anyways.